packages/hash/frontend/src/components/GlideGlid/utils/custom-grid-icons.ts

```diff
@@ -0,0 +1,28 @@
+import type { SpriteProps } from "@glideapps/glide-data-grid/dist/ts/common/utils";
+
+export enum CustomGridIcon {
```

Comment:
I'd suggest to avoid enums in the codebase. We can use discriminating types instead in most cases. `type X = "a" | "b" | "c"`. I used to work on a pretty large codebase with TS enums and it had issues with circular deps because enums act both as types and runtime.

Unless someone finds it critical to use `enums` in the codebase, I’d create a ESLint rule that permanently bans them:

https://github.com/kachkaev/routine-npm-packages/blob/2ce0f5322df4684da563afca0feda07d566cbe66/packages/eslint-config-base/index.cjs#L3-L7

Reply:
Replaced enum with union type here https://github.com/hashintel/hash/pull/1266/commits/ef012d6115cbdd6553817658f41f771e63294940